dev-tools/mage/dockerbuilder.go

```diff
@@ -178,6 +178,7 @@ func (b *dockerBuilder) expandDockerfile(templatesDir string, data map[string]in
 
 func (b *dockerBuilder) dockerBuild() (string, error) {
 	tag := fmt.Sprintf("%s:%s", b.imageName, b.Version)
+	tag = strings.Replace(tag, "+", "_", 1)
```

Comment:
https://github.com/opencontainers/distribution-spec/issues/154

I see a dash '-' used most frequently to replace the '+', this seems to be what OCI itself was doing before https://github.com/opencontainers/distribution-spec/pull/417/files

So I have a small preference for that just to avoid doing this in a way that isn't consistent with other projects that also have this problem.